tools/clang/test/CodeGenDXIL/operators/swizzle/indexSwizzledBoolVec.hlsl

```diff
@@ -0,0 +1,20 @@
+// Test indexing a swizzled bool vector
+// RUN: %dxc -T cs_6_0 %s | FileCheck %s
+
+// This was asserting in Instructions.cpp with:
+// void llvm::StoreInst::AssertOK(): Assertion `getOperand(0)->getType() == cast<PointerType>(getOperand(1)->getType())->getElementType() && "Ptr must be a pointer to Val type!"' failed.
+
+// CHECK: @main
```

Comment:
Checking that it doesn't crash isn't a good enough test. We tend to expect that inputs both don't crash _and_ produce valid output.

You should be able to add the `-fcgl` flag to get the pre-optimized IR, and do some checks against that to verify that the IR generation is correct.

Reply:
That's fair. I had based this on other examples I saw of hlsl tests that seem to validate that dxc doesn't crash. But a better test is a good thing, so I'll do this.

Reply:
Done.